### DISCUSSION TITLE
By pass git-hasher for e2e projects in tmpProjPath

### DISCUSSION THREAD

**Discussion by lewsmith** [nrwl/nx#10920] [General]
I've got an nx-plugin module with an e2e test that uses the `dep-graph` command to grab a json of changes.

The e2e has a test that contains roughly the following:

```
const appName = uniq('app')
const libName = uniq('lib')

ensureNxProject('@package/my-package', 'dist/packages/my-package')

await runNxCommandAsync(`generate @package/my-package:application ${appName}`)
await runNxCommandAsync(`generate @package/my-package:library ${libName}`)
await runNxCommandAsync(`generate @package/my-package:setup-my-package-plugin`)

await runNxCommandAsync('dep-graph --file=output.json')
const { graph } = readJson('output.json')

expect(graph).toBeDefined()
expect(graph.dependencies).toBeDefined()
expect(graph.dependencies[appName]).toBeDefined()

const appDependencies = graph.dependencies[appName]
expect(appDependencies.length).toBe(1)
expect(appDependencies[0].target).toBe(libName)
```

I do also have other tests in this suite that all work perfectly, however this one fails because the data being returned from the `dep-graph` command does not contain any dependencies. 

I've worked out that this is down to the fact that my root `.gitignore` file contains an entry for `/tmp`, and this in turn stops the `dep-graph` from working (both `context.filesMap` and `context.filesToProcess` is empty), because the e2e test creates the project under the `/tmp` directory when using `ensureNxProject`.

If I remove the `/tmp` entry in my root ignore file, everything works perfectly.

So I'm wondering if there Is there any way to bypass the git-hashing inside a test file? I can't seem to be able to disable the cache when calling `dep-graph`. I've tried adding another ignore file with `!tmp/` in the temporary project (before calling the dep-graph command) but it made no difference.

Any help would be appreciated

**Comment by klagrida** [comment]
@lewsmith Thanks, I have the same problem in my nx plugin and removing `/tmp` from workspace `gitignore` solved it. I believe this should be an issue to be fixed.
@AgentEnder what do you think please?
